sycl/source/detail/device_impl.hpp

```diff
@@ -1236,8 +1236,8 @@ class device_impl : public std::enable_shared_from_this<device_impl> {
     }
     CASE(emulated) { return false; }
     CASE(host_debuggable) { return false; }
-    CASE(fp16) { return has_extension("cl_khr_fp16"); }
-    CASE(fp64) { return has_extension("cl_khr_fp64"); }
```

Comment:
This relied on the internal caching of the extensions string, we need to ensure that new underlying query is cached too. See https://github.com/intel/llvm/blob/76a887eb43a2a441ae059601481ac88a38293974/sycl/source/detail/device_impl.hpp#L2265-L2301